app/src/main/java/lt/vilnius/tvarkau/fragments/ReportFilterFragment.kt

```diff
@@ -0,0 +1,147 @@
+            unSelectAllStates()
+            it.isSelected = true
+        }
+
+        filter_report_status_new.setOnClickListener { onSelectState(it) }
+        filter_report_status_registered.setOnClickListener { onSelectState(it) }
+        filter_report_status_completed.setOnClickListener { onSelectState(it) }
+    }
+
+    private fun unSelectAllStates() {
+        filter_report_status_new.isSelected = false
+        filter_report_status_registered.isSelected = false
+        filter_report_status_completed.isSelected = false
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        with(activity as BaseActivity) {
+            setSupportActionBar(toolbar)
+            setTitle(R.string.report_filter_page_title)
+
+            toolbar.setNavigationIcon(R.drawable.ic_close)
+            toolbar.setOnMenuItemClickListener {
+                when (it.itemId) {
+                    R.id.action_send -> {
+                        onSubmitFilter()
+                        activity.onBackPressed()
+                        true
+                    }
+                    else -> false
+                }
+            }
+            toolbar.setNavigationOnClickListener {
+                activity.onBackPressed()
+            }
+        }
+
+        val previouslySelectedStatus = reportStatusFilter.get()
+        if (previouslySelectedStatus.isNullOrEmpty()) {
+            filter_report_status_new.isSelected = true
+        } else {
+            filter_report_status_registered.isSelected = previouslySelectedStatus == STATUS_REGISTERED
+            filter_report_status_completed.isSelected = previouslySelectedStatus == STATUS_DONE
+        }
+
+        adapter = FilterReportTypesAdapter(
+                reportTypes,
+                selectedReportType,
+                {
+                    reportTypeFilter.set(it)
+                    adapter.selected = it
+                    adapter.notifyDataSetChanged()
+                }
+        )
+        filter_report_types.adapter = adapter
+
+        reportTypesInteractor.getReportTypes()
+                .observeOn(uiScheduler)
+                .subscribe({
+                    reportTypes.addAll(it)
+                    adapter.notifyDataSetChanged()
+                }, {
+
+                })
+    }
+
+    override fun onInject(component: ApplicationComponent) {
+        component.inject(this)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.submit_filter, menu)
+    }
+
+    private fun onSubmitFilter() {
+        val selectedState: String? = listOf(
+                filter_report_status_new,
+                filter_report_status_registered,
+                filter_report_status_completed
+        ).find { it.isSelected }?.tag as? String
+
+        reportStatusFilter.set(selectedState.nullToEmpty())
```

Comment:
Well, `StringPreference` is not tied directly to `SharedPreferences` and implementation could have any serializer. Database, file even network.
So we describe interface what is acceptable.